src/components/pages/lessons/overlay/go-pro-cta-overlay.tsx

```diff
@@ -2,30 +2,170 @@ import React, {FunctionComponent} from 'react'
 import Link from 'next/link'
 import {LessonResource} from 'types'
 import {track} from 'utils/analytics'
+import {useCommerceMachine} from 'hooks/use-commerce-machine'
+import {PlanPrice} from 'components/pricing/select-plan-new/index'
+import SmallParityCouponMessage from './small-parity-coupon-message'
+import {Coupon} from 'machines/commerce-machine'
+import get from 'lodash/get'
+import isEmpty from 'lodash/isEmpty'
+import first from 'lodash/first'
+import {useViewer} from 'context/viewer-context'
+import emailIsValid from 'utils/email-is-valid'
+import axios from 'axios'
+import stripeCheckoutRedirect from 'api/stripe/stripe-checkout-redirect'
+import toast from 'react-hot-toast'
+import {StripeAccount} from 'types'
 
 type JoinCTAProps = {
   lesson: LessonResource
 }
 
 const GoProCtaOverlay: FunctionComponent<JoinCTAProps> = ({lesson}) => {
+  const {viewer, authToken} = useViewer()
+  const {state, send, priceId, quantity, availableCoupons, currentPlan} =
+    useCommerceMachine({initialPlan: 'monthlyPrice'})
+
```

Comment:
A lot of the variables and derived data below is very similar to what we are doing on the pricing page, but there are minor differences to fit this CTA overlay. In particular, the `onClickCheckout` handler in particular.